leshan-client-core/src/main/java/org/eclipse/leshan/client/resource/RootEnabler.java

```diff
@@ -212,6 +213,12 @@ public WriteCompositeResponse write(ServerIdentity identity, WriteCompositeReque
     public synchronized ObserveCompositeResponse observe(ServerIdentity identity, ObserveCompositeRequest request) {
         List<LwM2mPath> paths = request.getPaths();
 
+        try {
+            validatePathsNotOverlapping(paths);
+        } catch (InvalidRequestException exception) {
+            return ObserveCompositeResponse.badRequest(exception.getMessage());
+        }
+
```

Comment:
I think this is not needed because `ObserveCompositeResponse.getPaths();` can not contain overlapped paths, because if you try to do that an `InvalidRequestException` will be raised.

Maybe it worth to double check but I think in client `RootResource` we will try to create the compositeRequest, the exception will be raised, and then catch and return a bad request. (see `LwM2mCoapResource.handleRequest`)

Reply:
You're right, but there's no way to be 100% sure, since server and client check share the same validator in `ObserveCompositeResponse`'s constructor. That being seid, if I remove the check in `RootEnabler`, it should still work.

Reply:
By double check, I meant that it worth you read this code too to be sure you understand like me.

If you understand like me, I think we can remove the code in `RootEnabler`.

Reply:
Yup, I understand it and I did remove the code from `RootEnabler`